docs/databases/swift-kuery.html

```diff
@@ -252,237 +229,150 @@ <h2 class="heading-2"><span class="blue-text">Step 5 (Optional):</span> Test sav
-                  let valueString = String(describing: value)
-                  resultString.append("\(valueString.padding(toLength: 35, withPad: " ", startingAt: 0))")
+                var books = [Book]()
+                resultSet.forEach() { row, error in
+                    guard let row = row else {
+                        if let error = error {
+                            Log.error("Error getting row: \(error)")
+                            return completion(nil, .internalServerError)
+                        } else {
+                            // All rows have been processed
+                            return completion(books, nil)
+                        }
+                    }
+                    guard let idString = row[0] as? String,
+                        let id = Int(idString),
+                        let title = row[1] as? String,
+                        let price = row[2] as? Double,
+                        let genre = row[3] as? String
+                    else {
+                            Log.error("Unable to decode book")
+                            return completion(nil, .internalServerError)
+                    }
+                    books.append(Book(id: id, title: title, price: price, genre: genre))
                 }
-              }
-              resultString.append("\n")
-          }
+            }
         }
-      }
     }
-  }</code></pre>
-      <p>We first call the `forEach()` function on the result set which gives us access to the row and a way to log errors.</p>
-      <p>Next we confirm we have a valid row, if not we log an error.</p>
-      <p>If the row is nil but error is not we assume the loop has completed and call the callback passing in `resultString`</p>
-      <p>Otherwise we loop over each row adding their values to the `resultString`.</p>
-      <div class="info">
-        <p>We use the padding method to create a more readable output in the browser.</p>
-        <p>For the same reason we also append a `"\n"` which is a break line.</p>
-      </div>
-      <p>That's it! We've setup our GET route to retrieve data from a database.</p>
+}</code></pre>
       <div class="underline"></div>
       <h2 class="heading-2"><span class="blue-text">Step 7 (Optional):</span> Test retrieving from a database</h2>
-      <div class="info">
-        <p>If you followed Step 6 (Optional): Test saving to database then you can simply start your server and navigate to: <a target="_blank" href="http://localhost:8080/books">http://localhost:8080/books</a></p>
-        <p>You should see the following:</p>
-        <pre><code> id |       title       | price |  genre
-  ----+-------------------+-------+---------
-   0  | A Game of Thrones | 14.99 | Fantasy
-  (1 row)</code></pre>
-      </div>
+      <p>If you followed Step 6 (Optional): Test saving to database then you can simply start your server and navigate to: <a target="_blank" href="http://localhost:8080/kuery">http://localhost:8080/kuery</a></p>
```

Comment:
Test saving to database is now Step 5.
Is the step 7 title meant to be Test retrieving data from a database?
Step 6 is not marked optional - is it meant to be if steps 5 and 7 are?